token/js/src/extensions/transferFee/state.ts

```diff
@@ -41,6 +41,19 @@ export function transferFeeLayout(property?: string): Layout<TransferFee> {
     return struct<TransferFee>([u64('epoch'), u64('maximumFee'), u16('transferFeeBasisPoints')], property);
 }
 
+/** Calculate the transfer fee */
+export function calculateFee(transferFee: TransferFee, preFeeAmount: bigint): bigint {
+    const transferFeeBasisPoints = transferFee.transferFeeBasisPoints;
+    if (transferFeeBasisPoints === 0 || preFeeAmount === BigInt(0)) {
+        return BigInt(0);
+    } else {
+        const numerator = preFeeAmount * BigInt(transferFeeBasisPoints);
+        const rawFee = numerator / ONE_IN_BASIS_POINTS;
```

Comment:
Sorry, I'm just now realizing that this is incorrect. We do a "ceiling" division in the rust side, so that it always gives the "ceiling" result, and not the floor like this code is doing. To get the ceiling, you can do:
```suggestion
        const rawFee = (numerator + ONE_IN_BASIS_POINTS - 1) / ONE_IN_BASIS_POINTS;
```

See https://github.com/solana-labs/solana-program-library/blob/2a2b2a763b0847ebe6a028016b14d04d95bf655d/token/program-2022/src/extension/transfer_fee/mod.rs#L38-L62 for more info

Note that this change will modify the results from your tests